2022Swerve/src/main/java/frc/robot/subsystems/AutonSubsytem.java

```diff
@@ -0,0 +1,137 @@
+        yController = new PIDController(AutonConstants.kYTranslationP, 0, 0);
+        rotationController = new ProfiledPIDController(
+            AutonConstants.kRotationP, 0, 0, AutonConstants.kRotationalConstraints);
+        rotationController.enableContinuousInput(-Math.PI, Math.PI);
+    }
+
+    private Trajectory getTrajectory(String name){
+        Trajectory trajectory;
+        try {
+            Path trajectoryPath = Filesystem.getDeployDirectory().toPath().resolve("pathplanner/generatedJSON/" + name + ".wplib.json");
+            trajectory = TrajectoryUtil.fromPathweaverJson(trajectoryPath);
+        } catch (IOException ex) {
+            DriverStation.reportError("Unable to open trajectory: " + name, ex.getStackTrace());
+            trajectory = null;
+        }
+        return trajectory;
+    }
+
+    private SwerveControllerCommand followTrajectory(String name){
+        trajectoryLog.append(name);
+        return new SwerveControllerCommand(
+            getTrajectory(name),
+            swerveSubsystem::getPose, 
+            DriveConstants.kDriveKinematics, 
+            xController, 
+            yController, 
+            rotationController, 
+            swerveSubsystem::setModuleStates, 
+            swerveSubsystem
+        );
+    }
+
+    private Command resetOdometry(String initialTrajectory){
+        //Resets odometry to the initial position of the given trajectory
+        return new InstantCommand(() -> swerveSubsystem.resetOdometry(getTrajectory(initialTrajectory).getInitialPose()));
+    }
+    
+    private Command getAutonSequence(){
+        autonMode = modeChooser.getSelected();
+        //Sequence of actions to be performed during the autonomous period
+        switch(autonMode){
+            case FORWARD:
+               return 
+                    new SequentialCommandGroup(
+                        resetOdometry("Forward"),
+                        followTrajectory("Forward")
+                    );
+            case BACKWARD:
+                return
+                    new SequentialCommandGroup(
+                        resetOdometry("Backward"),
+                        new WaitCommand(3),
+                        followTrajectory("Backward")
+                    );
+            case FORWARD180:
+                return
+                    new SequentialCommandGroup(
+                        resetOdometry("Forward180"),
+                        followTrajectory("Forward180")
+                    );
+            default:
+                return null;
+        }
+    }
+
+    private Command getAutonEnd(){
+        //Actions to be performed unconditionally after the autonomous seqence has ended (Stop motors)
+        return new SequentialCommandGroup(
+            new InstantCommand(() -> swerveSubsystem.stopModules())
+        );
+    }
+
+    public Command getAutonCommand(){
+        return getAutonSequence().beforeStarting(new WaitCommand(delayEntry.getDouble(0.0))).andThen(getAutonEnd());
```

Comment:
Check last year's code, did we have more than one delay point?  What's your plan for that?

Do you think that it would make it more straightforward to have an autoDelay method that returns a new WaitCommand?  
Not sure.

Reply:
( I see that you have a new waitCommand in BACKWARD now )

Reply:
This is exclusively an initial wait time to be set by shuffleboard, like last year's delay functionality. Any time we need a delay in a sequence, we can just add a wait command into that sequence.

Reply:
k.  And you mentioned a specific wait command, closing comment